stravalib/client.py

```diff
@@ -1815,7 +1920,15 @@ def delete_subscription(self, subscription_id, client_id, client_secret):
         # Expects a 204 response if all goes well.
 
 
-class BatchedResultsIterator(object):
+T = TypeVar("T", bound=BaseModel)
+
+
+class ResultFetcher(Protocol):
+    def __call__(self, *, page: int, per_page: int) -> Iterable[Any]:
+        ...
+
+
+class BatchedResultsIterator(Generic[T]):
```

Comment:
It's great that you turned this into a generic class 👍

Reply:
That really helps get good typing of the method that uses it 🙂